### PR TITLE
upgrade client-go package from v0.17.1 to v0.19.10,

### PR DESCRIPTION
**Pull request by liufen90**
upgrade client-go package from v0.17.1 to v.0.19.10, #214

### CODE REVIEW

**Comment by cheftako** [comment]
/lgtm
Have you validated that this works with older Kubernetes APIServers (prior to v1.19?)
If not we may need to implement a README for our releases.

**Comment by liufen90** [comment]
> /lgtm
> Have you validated that this works with older Kubernetes APIServers (prior to v1.19?)
> If not we may need to implement a README for our releases.

No, I haven't.

**Comment by cheftako** [comment]
@caesarxuchao may know if this is a concern. I don't think its worth gating this PR on this but it would be nice to know if we should have a README for it. Waiting on Chao's review to approve.

**Comment by liufen90** [comment]
@caesarxuchao please take a look. Thank you.

File: pkg/server/server.go [hunk 1]
```diff
@@ -508,14 +509,14 @@ func getAgentIdentifiers(stream agent.AgentService_ConnectServer) (pkgagent.Iden
...
-func (s *ProxyServer) validateAuthToken(token string) error {
+func (s *ProxyServer) validateAuthToken(ctx context.Context, token string) error {
 	trReq := &authv1.TokenReview{
 		Spec: authv1.TokenReviewSpec{
 			Token:     token,
 			Audiences: []string{s.AgentAuthenticationOptions.AuthenticationAudience},
 		},
 	}
-	r, err := s.AgentAuthenticationOptions.KubernetesClient.AuthenticationV1().TokenReviews().Create(trReq)
+	r, err := s.AgentAuthenticationOptions.KubernetesClient.AuthenticationV1().TokenReviews().Create(ctx, trReq, metav1.CreateOptions{})
```
Comment (caesarxuchao) [inline pkg/server/server.go:519]: This is the only use of client-go in the code base. I think this client call works for older apiservers.

**Review by caesarxuchao — COMMENTED** [review]
/lgtm

**Comment by cheftako** [comment]
For reference its in tag v0.0.19 (Coincidence).

**Pull request merged**